src/stream.jl

```diff
@@ -229,50 +229,26 @@ end
     end
 end
 
-function Base.ismarked(stream::TranscodingStream)
+function Base.ismarked(stream::TranscodingStream)::Bool
     checkmode(stream)
-    return ismarked(stream.buffer1)
+    isopen(stream) && ismarked(stream.buffer1)
 end
 
-function Base.mark(stream::TranscodingStream)
-    checkmode(stream)
-    return mark!(stream.buffer1)
+function Base.mark(stream::TranscodingStream)::Int64
+    ready_to_read!(stream)
+    mark!(stream.buffer1)
+    position(stream)
 end
 
-function Base.unmark(stream::TranscodingStream)
+function Base.unmark(stream::TranscodingStream)::Bool
     checkmode(stream)
-    return unmark!(stream.buffer1)
+    isopen(stream) && unmark!(stream.buffer1)
 end
 
-function Base.reset(stream::TranscodingStream)
-    checkmode(stream)
-    return reset!(stream.buffer1)
-end
-
-function Base.skip(stream::TranscodingStream, offset::Integer)
-    checkmode(stream)
-    if offset < 0
-        throw(ArgumentError("negative offset"))
-    end
-    mode = stream.state.mode
-    buffer1 = stream.buffer1
-    skipped = 0
-    if mode === :read || mode === :stop
-        while !eof(stream) && buffersize(buffer1) < offset - skipped
-            n = buffersize(buffer1)
-            emptybuffer!(buffer1)
-            skipped += n
-        end
-        if eof(stream)
-            emptybuffer!(buffer1)
-        else
-            skipbuffer!(buffer1, offset - skipped)
-        end
-    else
-        # TODO: support skip in write mode
```

Comment:
I'm not sure what this TODO means, but I add a docstring to clarify that `skip` behaves like a read that ignores `eof` and discards the read bytes.

Reply:
I also moved the `skip` to be near the other read functions